modules/Behaviour/behaviour_manage_add.php

```diff
@@ -179,11 +183,8 @@
                 $page->addError(__('You have not specified one or more required parameters.'));
             } else {
                 //Check for existence of behaviour record
-
-                    $data = array('gibbonSchoolYearID' => $session->get('gibbonSchoolYearID'), 'gibbonBehaviourID' => $gibbonBehaviourID);
-                    $sql = "SELECT * FROM gibbonBehaviour JOIN gibbonPerson ON (gibbonBehaviour.gibbonPersonID=gibbonPerson.gibbonPersonID) JOIN gibbonStudentEnrolment ON (gibbonPerson.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID) JOIN gibbonFormGroup ON (gibbonStudentEnrolment.gibbonFormGroupID=gibbonFormGroup.gibbonFormGroupID) WHERE gibbonFormGroup.gibbonSchoolYearID=:gibbonSchoolYearID AND status='Full' AND (dateStart IS NULL OR dateStart<='".date('Y-m-d')."') AND (dateEnd IS NULL  OR dateEnd>='".date('Y-m-d')."') AND gibbonBehaviourID=:gibbonBehaviourID";
-                    $result = $connection2->prepare($sql);
-                    $result->execute($data);
+                $result = $container->get(BehaviourGateway::class)->getBehaviourRecordToCheck($session->get('gibbonSchoolYearID'), $gibbonBehaviourID);
```

Comment:
Behaviour is attached to a school year, so passing the gibonSchoolYearID in is unnecessary (blame Ross 😅). You can join the gibbonStudentEnrolment table on the school year from the gibbonBehaviour taable.
```suggestion
                $result = $container->get(BehaviourGateway::class)->getBehaviourRecordByID($gibbonBehaviourID);
```